File: MetaBrain/imaging/feature-flags.ts
import type { ImageFeatureFlags } from "./types";

export const DEFAULT_IMAGE_FEATURE_FLAGS: ImageFeatureFlags = {
  medicalVisionEnabled: false,
  medicalVisionShadowMode: true,
  medicalVisionProviderEnabled: false,
  dicomEnabled: false,
  dicomShadowMode: true,
  imageHumanReviewRequired: true,
  imageStoreOriginal: false,
};

const TRUE_VALUES = new Set(["1", "true", "yes", "on"]);
const FALSE_VALUES = new Set(["0", "false", "no", "off"]);

function readBoolean(value: string | undefined, fallback: boolean): boolean {
  if (value === undefined) {
    return fallback;
  }
  const normalized = value.trim().toLowerCase();
  if (TRUE_VALUES.has(normalized)) {
    return true;
  }
  if (FALSE_VALUES.has(normalized)) {
    return false;
  }
  return fallback;
}

export function loadImageFeatureFlags(env: Record<string, string | undefined> = process.env): ImageFeatureFlags {
  return {
    medicalVisionEnabled: readBoolean(env.MEDICAL_VISION_ENABLED, DEFAULT_IMAGE_FEATURE_FLAGS.medicalVisionEnabled),
    medicalVisionShadowMode: readBoolean(
      env.MEDICAL_VISION_SHADOW_MODE,
      DEFAULT_IMAGE_FEATURE_FLAGS.medicalVisionShadowMode,
    ),
    medicalVisionProviderEnabled: readBoolean(
      env.MEDICAL_VISION_PROVIDER_ENABLED,
      DEFAULT_IMAGE_FEATURE_FLAGS.medicalVisionProviderEnabled,
    ),
    dicomEnabled: readBoolean(env.DICOM_ENABLED, DEFAULT_IMAGE_FEATURE_FLAGS.dicomEnabled),
    dicomShadowMode: readBoolean(env.DICOM_SHADOW_MODE, DEFAULT_IMAGE_FEATURE_FLAGS.dicomShadowMode),
    imageHumanReviewRequired: readBoolean(
      env.IMAGE_HUMAN_REVIEW_REQUIRED,
      DEFAULT_IMAGE_FEATURE_FLAGS.imageHumanReviewRequired,
    ),
    imageStoreOriginal: readBoolean(env.IMAGE_STORE_ORIGINAL, DEFAULT_IMAGE_FEATURE_FLAGS.imageStoreOriginal),
  };
}

export const DOCUMENTED_IMAGE_FLAGS = {
  MEDICAL_VISION_ENABLED: "false",
  MEDICAL_VISION_SHADOW_MODE: "true",
  MEDICAL_VISION_PROVIDER_ENABLED: "false",
  DICOM_ENABLED: "false",
  DICOM_SHADOW_MODE: "true",
  IMAGE_HUMAN_REVIEW_REQUIRED: "true",
  IMAGE_STORE_ORIGINAL: "false",
} as const;

File: MB-Secretaria/imaging_py/feature_flags.py
from __future__ import annotations

import os
from typing import Optional

from .types import ImageFeatureFlags


def _read_bool(value: Optional[str], fallback: bool) -> bool:
    if value is None:
        return fallback
    normalized = value.strip().lower()
    if normalized in {"1", "true", "yes", "on"}:
        return True
    if normalized in {"0", "false", "no", "off"}:
        return False
    return fallback


def load_image_feature_flags(env: dict[str, str] | None = None) -> ImageFeatureFlags:
    values = env if env is not None else os.environ
    return ImageFeatureFlags(
        medical_vision_enabled=_read_bool(values.get("MEDICAL_VISION_ENABLED"), False),
        medical_vision_shadow_mode=_read_bool(values.get("MEDICAL_VISION_SHADOW_MODE"), True),
        medical_vision_provider_enabled=_read_bool(values.get("MEDICAL_VISION_PROVIDER_ENABLED"), False),
        dicom_enabled=_read_bool(values.get("DICOM_ENABLED"), False),
        dicom_shadow_mode=_read_bool(values.get("DICOM_SHADOW_MODE"), True),
        image_human_review_required=_read_bool(values.get("IMAGE_HUMAN_REVIEW_REQUIRED"), True),
        image_store_original=_read_bool(values.get("IMAGE_STORE_ORIGINAL"), False),
    )
